src/main/scala/com/cibo/scalastan/StanType.scala

```diff
@@ -302,18 +317,28 @@ case class StanArray[CONTAINED <: StanType] private[scalastan] (
       data.length +: inner.getDims(Seq.empty.asInstanceOf[inner.SCALA_TYPE])
     }
   }
-  private[scalastan] def parse(name: String, values: Map[String, String]): SCALA_TYPE = {
+  private[scalastan] def parse(
+    name: String,
+    parameterChains: Map[String, Vector[Vector[String]]]
+  ): Vector[Vector[SCALA_TYPE]] = {
     // The name will be the prefix up to the number we need to parse.
     val prefix = s"$name."
     val prefixLength = prefix.length
 
     // Get the size of this dimension (note that arrays in STAN are 1-based).
-    val size = values.keys.filter(_.startsWith(prefix)).map(_.drop(prefixLength).takeWhile(Character.isDigit).toInt).max
+    val size = parameterChains.keys.filter(_.startsWith(prefix)).map(
+      _.drop(prefixLength).takeWhile(Character.isDigit).toInt
+    ).max
 
     // Parse the next level for each element of this array.
-    Vector.tabulate[CONTAINED#SCALA_TYPE](size) { i =>
+    val allValues = Vector.tabulate[Vector[Vector[CONTAINED#SCALA_TYPE]]](size) { i =>
       val nextName = s"$prefix${i + 1}"
-      inner.parse(nextName, values)
+      inner.parse(nextName, parameterChains)
+    }
+    parameterChains.head._2.zipWithIndex.map { case (chain, chainIndex) =>
```

Comment:
more of a question for Joe, but this `parameterChains.head._1.zipWithIndex.map({case(cahin, chainIndex...` bit shows up in a couple of places so I thought I'd ask.

Is this the best way of creating an output with the same size as the samples of any given variable?  My scala collections-fu isn't good enough to say, but this caught my eye and I wanted to ask.

Reply:
Looks reasonable to me.  It could probably be simplified if we assume all chains are the same size (I’d like to avoid that assumption as much as possible though just in case).  As you mention, probably good to pull this out somehow if it’s in several places so that we can change it more easily if desired.